packages/terra-table/src/SelectableTableRows.jsx

```diff
@@ -31,24 +35,77 @@ const propTypes = {
 const defaultProps = {
   disableUnselectedRows: false,
   onChange: undefined,
+  onBlur: undefined,
   selectedIndexes: [],
 };
 
+const rowsSelector = 'row-group';
+
 class SelectableTableRows extends React.Component {
   constructor(props) {
     super(props);
     this.handleOnChange = this.handleOnChange.bind(this);
+    this.handleOnBlur = this.handleOnBlur.bind(this);
+  }
+
+
+  componentDidMount() {
+    // Adds ARIA-Label for Pre-selected rows
+    const rowGroup = document.querySelector(`#${rowsSelector}`);
+    for (let rowIndex = 0; rowIndex < rowGroup.children.length; rowIndex += 1) {
+      const preselectedRow = rowGroup.children[rowIndex];
+      let ariaLabel = 'pre-selected row';
+      if ((preselectedRow.hasAttribute('aria-selected')) && (preselectedRow.getAttributeNode('aria-selected').value === 'true')) {
+        for (let columnIndex = 0; columnIndex < preselectedRow.children.length; columnIndex += 1) {
+          ariaLabel += ` ${preselectedRow.children[columnIndex].innerText}`;
+        }
+        rowGroup.children[rowIndex].setAttribute('aria-label', ariaLabel);
+      }
+    }
   }
 
   handleOnChange(event, index) {
     if (this.props.onChange) {
       this.props.onChange(event, index);
     }
+
+    if (!document.activeElement.hasAttribute('aria-label')) {
+      // Aria-Label is being populated with the cell contents for screen reader to read it as whole instead of cell value.
+      const ariaLabel = 'row selected';
```

Comment:
If this is the text the screen reader will announce it will need to be translated.

This PR is a little large, but it has good examples of dynamically announcing selections: https://github.com/cerner/terra-core/pull/2236

Reply:
added translations for  [on click](https://github.com/cerner/terra-core/blob/cc6ee3fad6bea01cc24bb5ce812e6483291d781d/packages/terra-table/src/SelectableTableRows.jsx#L67) and for [pre selected row](https://github.com/cerner/terra-core/blob/cc6ee3fad6bea01cc24bb5ce812e6483291d781d/packages/terra-table/src/SelectableTableRows.jsx#L172)